oracle/src/feeds/dia_fair_price.rs

```diff
@@ -0,0 +1,122 @@
+#![allow(clippy::single_char_pattern)]
+use super::{get_http, PriceFeed};
+use crate::{config::CurrencyStore, currency::*, Error};
+use async_trait::async_trait;
+use clap::Parser;
+use reqwest::Url;
+use serde::Deserialize;
+use serde_json::Value;
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct DiaFairPriceExt {
+    alias: Option<String>,
+}
+
+#[derive(Parser, Debug, Clone)]
+pub struct DiaFairPriceCli {
+    /// Fetch the exchange rate from Dia xLSD feed
+    #[clap(long)]
+    dia_fair_price_url: Option<Url>,
+}
+
+pub struct DiaFairPriceApi {
+    url: Url,
+}
+
+impl Default for DiaFairPriceApi {
+    fn default() -> Self {
+        Self {
+            url: Url::parse("https://api.diadata.org/xlsd/").unwrap(),
+        }
+    }
+}
+
+fn extract_response(value: Value, alias: &str) -> Option<f64> {
+    value
+        .as_array()?
+        .into_iter()
+        .find(|entry| matches!(entry.get("Token").and_then(|value| value.as_str()), Some(token) if token.to_uppercase() == alias))?
```

Comment:
should we do alias to uppercase as well?

Reply:
The oracle config already assumes all token symbols are capitalized, if this is not the case then many parts of the code would not work correctly.